src/Microsoft.AspNetCore.OData/Batch/ODataBatchResponseItem.cs

```diff
@@ -17,6 +17,51 @@ namespace Microsoft.AspNet.OData.Batch
     /// </summary>
     public abstract class ODataBatchResponseItem
     {
+        /// <summary>
+        /// Writes a single OData batch response Synchronously.
+        /// </summary>
+        /// <param name="writer">The <see cref="ODataBatchWriter"/>.</param>
+        /// <param name="context">The message context.</param>
+        public static void WriteMessage(ODataBatchWriter writer, HttpContext context)
```

Comment:
>atic void WriteMessage(ODataBatchWriter writer, HttpContext con [](start = 17, length = 63)

can we call "WriteMessageAsync" in WriteMessage, or Call "WriteMessage()" in "WriteMessageAsync"?

Reply:
I got rid of WriteMessage, and instead just created a WriteMessageAsync that took a flag to determine whether the writer was async.  When we have a major breaking change release of WebAPI OData I want to make all of this internal; there's no need for this functionality to be public, and it's confusing and broken in its current state.

---
In reply to: [378431693](https://github.com/OData/WebApi/pull/2048#discussion_r378431693) [](ancestors = 378431693)